[2.0] Browser history feature clean up

inxilpro opened the pull request:
I took one pass over the browser history implementation and cleaned things up a bit:

- Minor tweak to JS
- Renamed `getFromQueryString` to `getQueryString` to keep naming consistent with the rest of the component API
- Removed `$casts` property and `getCasts()` method because that functionality was removed
- Removed leftover call to an `AllCapsCaster` class
- Refactored `SupportBrowserHistory` for consistent naming (there was still a mix of `$queryStringParams` vs `$queryStringParams` vs `$queryParams` throughout)
- Other general refactors to `SupportBrowserHistory` to remove unnecessary duplication/etc

inxilpro committed:
Clean up

timrspratt commented:
👍 This should hopefully resolve an issue I'm currently having where Livewire is adding `?` onto the url despite no query params being present when using:

```php
protected $queryString = [
    'search' => ['except' => ''],
];
```

inxilpro committed:
Additional test

Add test case for https://github.com/livewire/livewire/pull/1476#issuecomment-687862154

inxilpro commented:
Cool. I just added a test for that specific case just to be safe :)

inxilpro committed:
Merge remote-tracking branch 'upstream/master' into feature/history-review

calebporzio commented:
Thanks @inxilpro, this is great!
